### PR TITLE
add "allowed label changes" policy

### LINKED ISSUE #247: [Sample] Allow "safe" labels to be added/changed/removed
### Problem Statement

In some cases, users are not permitted to change objects but, for operational emergencies, may be allowed to change one or more labels on those objects.

### Solution Description

Write a policy that permits users to add/change/remove a label with only a specific key to different types of resources.

This sample works:

```yaml
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: allowed-label-changes
  annotations:
    pod-policies.kyverno.io/autogen-controllers: none
spec:
  validationFailureAction: enforce
  background: false
  rules:
  - name: safe-label
    match:
      any:
      - resources:
          kinds:
          - Pod
          - Deployment
          - StatefulSet
          - DaemonSet
          - Job
          - CronJob
    preconditions:
      all:
      - key: "{{ request.operation }}"
        operator: Equals
        value: UPDATE
    validate:
      message: "The only label that may be removed or changed is `breakglass`."
      deny:
        conditions:
          any:
            - key: "{{ request.object.metadata.labels || `{}` |  merge(@, {breakglass:null}) }}"
              operator: NotEquals
              value: "{{ request.oldObject.metadata.labels || `{}` |  merge(@, {breakglass:null}) }}"
```

### Example "Good" Resource

_No response_

### Example "Bad" Resource

_No response_

### Other Comments

_No response_

### Slack discussion

_No response_

### Troubleshooting

- [X] I have read and followed the documentation AND the [troubleshooting guide](https://kyverno.io/docs/troubleshooting/).
- [X] I have verified the policy does not exist in the [samples library](https://kyverno.io/policies/).

### PR DESCRIPTION
**Pull request by chipzoller**
Signed-off-by: Chip Zoller <chipzoller@gmail.com>

## Related Issue(s)

Closes #247 
## Description

Adds a new sample policy called Allowed Label Changes
## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.

### CODE REVIEW

**Review by samj1912 — APPROVED** [review]

File: other/allowed_label_changes/allowed-label-changes.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      all:
+      - key: "{{ request.operation }}"
+        operator: Equals
+        value: UPDATE
+    validate:
+      message: "The only label that may be removed or changed is `breakglass`."
+      deny:
+        conditions:
+          any:
+            - key: "{{ request.object.metadata.labels || `{}` |  merge(@, {breakglass:null}) }}"
```
Comment (samj1912) [inline other/allowed_label_changes/allowed-label-changes.yaml:45]:
```suggestion
            - key: "{{ request.object.metadata.labels || `{}` |  merge(@, `{breakglass:null}`) }}"
```

File: other/allowed_label_changes/allowed-label-changes.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+              operator: NotEquals
+              value: "{{ request.oldObject.metadata.labels || `{}` |  merge(@, {breakglass:null}) }}"
```
Comment (samj1912) [inline other/allowed_label_changes/allowed-label-changes.yaml:47]:
```suggestion
              value: "{{ request.oldObject.metadata.labels || `{}` |  merge(@, `{breakglass:null}`) }}"
```

**Review by samj1912 — APPROVED** [review]

**Comment by samj1912** [comment]
@chipzoller what is the general contribution guideline for policies? Do we typically just add policies or do we add policies + test? I see a mixture of both in the repo.

**Comment by chipzoller** [comment]
> @chipzoller what is the general contribution guideline for the policies repo? Do we typically just add policies or do we add policies + test? I see a mixture of both in the repo.

We aren't mandating a new policy also have tests, but I'd like to get there. The problem remains that the CLI still has gaps that doesn't allow for proper testing (ex., no support for generate, no support for roles, clusterRoles, and subjects, etc.), which we must close ASAP. But also, speaking personally, when I'm in a time crunch, the highest priority is getting the sample policy out. I usually come around later and write tests.

**Comment by chipzoller** [comment]
> ```
> echo '{"breakglass": 1, "asd": 2}' | kyverno jp '@ || `{"a": 1}` | merge(@, {breakglass:null})'
> {
>   "asd": 2,
>   "breakglass": null
> }
> ```
> 
> LGTM

Yeah, the problem is enclosing `{breakglass:null}` in backticks causes Kyverno to throw 

```
Error from server: error when creating "testcpol.yaml": admission webhook "validate-policy.kyverno.svc" denied the request: policy contains invalid variables: variable substitution failed for rule safe-label: failed to resolve request.object.metadata.labels || `{}` |  merge(@, `{breakglass:null}`) at path /validate/deny/conditions/any/0/key: invalid JMESPath query request.object.metadata.labels || `{}` |  merge(@, `{breakglass:null}`): invalid character 'b' looking for beginning of object key string
```

**Pull request merged**